File: playwright/PageObjects/PagesPage.js
module.exports = class PagesPage {
  constructor(page) {
    this.page = page;
  }

  async goToCreateNewPage() {
    return this.page.click('a[href="#/editor/page/"]');
  }

  async openPageTypeFilterDropdown() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-selected-item:nth-child(1)');
  }

  async selectFilterByPublishedPagesOption() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-option:has-text("Published pages")');
  }

  async selectFilterByScheduledPagesOption() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-option:has-text("Scheduled pages")');
  }

  async clickPageWithTitle(title) {
    return this.page.click(`.gh-content-entry-title:has-text("${title}")`);
  }

  async selectFilterByDraftedPagesOption() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-option:has-text("Draft pages")');
  }

  async getFirstPageTitle() {
    await new Promise(r => setTimeout(r, 1000));
    const pagesTitles = await this.page.$$(".gh-content-entry-title");
    return pagesTitles[0].innerText();
  }

  async openPageTagsFilterDropdown() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('css=div.gh-contentfilter-menu.gh-contentfilter-tag');
  }

  async selectFilterByTagName(nameTag) {
    await new Promise(r => setTimeout(r, 1000));
    await this.page.click(`li:text("${nameTag}")`);
  }

  async searchPageByName(namePage) {
    await new Promise(r => setTimeout(r, 1000));
    const pageTitles = await this.page.$$("h3.gh-content-entry-title");
    for (var i = 0; i < pageTitles.length; i++) {
      if (await pageTitles[i].innerText() === namePage) {
        return true;
      }
    }
    return false;
  }

  async openPageSortByFilterDropdown() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-selected-item:has-text("Newest")');
  }

  async selectFilterByNewestPageOption() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-option:has-text("Newest")');
  }

  async selectFilterByRecentlyUpPagesOption() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-option:has-text("Recently updated")');
  }
};

File: kraken/features/web/step_definitions/page_objects/pages_page.rb
class PagesPage
  attr_reader :driver

  def initialize(driver)
    @driver = driver
  end

  def go_to_create_new_page
    @driver.find_element(
      :css, 'a[href="#/editor/page/"]'
    ).click
  end

  def open_page_type_filter_dropdown
    @driver.find_elements(
      :css, '.ember-power-select-selected-item'
    ).first.click
  end

  def select_filter_by_published_pages_option
    @driver.find_elements(
      :css, '.ember-power-select-option'
    ).select do |element|
      element.text == 'Published pages'
    end.first.click
  end

  def select_filter_by_drafted_pages_option
    @driver.find_elements(
      :css, '.ember-power-select-option'
    ).select do |element|
      element.text == 'Draft pages'
    end.first.click
  end

  def click_page_with_title(title)
    @driver.find_elements(
      :css, '.gh-content-entry-title'
    ).select do |element|
      element.text == title
    end.first.click
  end
end
